code/modules/mob/living/simple_animal/hostile/goose.dm

```diff
@@ -25,33 +25,122 @@
+	response_help  = "pets"
+	response_disarm = "gently pushes aside"
+	response_harm   = "kicks"
+	gold_core_spawnable = NO_SPAWN
+	random_retaliate = FALSE
+	var/vomiting = FALSE
+	var/vomitCoefficient = 1
+	var/vomitTimeBonus = 0
+	var/datum/action/cooldown/vomit/goosevomit
 
-/mob/living/simple_animal/hostile/retaliate/goose/toggle_ai(togglestatus)
+/mob/living/simple_animal/hostile/retaliate/goose/vomit/Initialize()
 	. = ..()
-	if(!key)
-		if(AIStatus != AI_ON)
-			set_resting(TRUE)
-		else
-			set_resting(FALSE)
+	goosevomit = new()
+	goosevomit.Grant(src)
 
-/mob/living/simple_animal/hostile/retaliate/goose/update_resting()
+/mob/living/simple_animal/hostile/retaliate/goose/vomit/Destroy()
+	QDEL_NULL(goosevomit)
+	return ..()
+
+/mob/living/simple_animal/hostile/retaliate/goose/vomit/examine(user)
 	. = ..()
-	if(resting)
-		wander = FALSE
-	else
-		wander = TRUE
-	update_icon()
+	to_chat(user, "<span class='notice'>Somehow, it still looks hungry.</span>")
 
-/mob/living/simple_animal/hostile/retaliate/goose/handle_automated_movement()
+/mob/living/simple_animal/hostile/retaliate/goose/vomit/attacked_by(obj/item/O, mob/user)
 	. = ..()
-	if(prob(5))
-		Retaliate()
+	feed(O)
 
-/mob/living/simple_animal/hostile/retaliate/goose/proc/update_icon()
-	if(stat != DEAD)
-		icon_state = resting ? icon_resting : icon_living
+/mob/living/simple_animal/hostile/retaliate/goose/vomit/proc/feed(obj/item/O)
+	. = ..()
+	if(!istype(O, /obj/item/reagent_containers/food))
+		return
+	var/obj/item/reagent_containers/food/tasty = O
+	if (tasty.foodtype & GROSS)
+		visible_message("<span class='notice'>[src] hungrily gobbles up \the [tasty]!</span>")
+		tasty.forceMove(src)
+		playsound(src,'sound/items/eatfood.ogg', 70, 1)
+		vomitCoefficient ++
+		vomitTimeBonus += 5
+	else
+		visible_message("<span class='notice'>[src] refuses to eat \the [tasty].</span>")
+
+/mob/living/simple_animal/hostile/retaliate/goose/vomit/proc/vomit()
+	var/turf/T = get_turf(src)
+	var/obj/item/reagent_containers/food/consumed = locate() in contents //Barf out a single food item from our guts
+	if (prob(50) && consumed)
+		barf_food(consumed)
 	else
-		icon_state = icon_dead
+		playsound(get_turf(src), 'sound/effects/splat.ogg', 50, 1) //yes getting the turf twice is necessary fuck off
+		T.add_vomit_floor(src)
+
+/mob/living/simple_animal/hostile/retaliate/goose/vomit/proc/barf_food(var/atom/A)
+	if(istype(A, /obj/item/reagent_containers/food))
+		return
+	var/turf/T1 = get_turf(src) //its not technically a one-letter var, checkmate ninjanom
+	var/obj/item/reagent_containers/food/consumed = A
+	consumed.forceMove(T1)
+	var/destination = get_edge_target_turf(T1, pick(GLOB.alldirs)) //Pick a random direction to toss them in
+	consumed.throw_at(destination, 1, 2) //Thow the food at a random tile 1 spot away
+	var/turf/T2 = get_turf(consumed)
+	T2.add_vomit_floor(src)
+	playsound(get_turf(consumed), 'sound/effects/splat.ogg', 50, 1) //yes getting the turf twice is necessary fuck off
```

Comment:
No it isn't?

Reply:
i tried editing it out and the sound wouldn't play, but i put it back and it worked again

Reply:
playsound(T2, the rest)

Reply:
yes i did that and it didnt play, no runtime error either

Reply:
I'll be testing this

Reply:
You shouldn't need this